opm/grid/CpGrid.hpp

```diff
@@ -80,6 +80,9 @@
 #include <list>
 #endif
 
+void refine_and_check(const Dune::cpgrid::Geometry<3, 3>&,
+                      const std::array<int, 3>&,
+                      bool);
```

Comment:
Hm, a new function in the global namespace?  Do we really need that?

Reply:
Thanks for noticing
I guess I have introduced this. This function is defined in tests/cpgrid/geometry_test.cpp. As we need to access private members there, we need it to be a friend. Hence the forward declaration here. Now that I reread this I am wondering whether the declaration is needed or the friend declaration inside of CpGrid suffices. I am always struggeling with friends if they come together with template parameters.

Reply:
Indeed, this forward declaration is unnecessary and we should remove it,